### PR TITLE
[USM] add missing map dumps

### PR DESCRIPTION
**Pull request by Yumasi**
### What does this PR do?

This PR adds a missing map dump to the debug endpoint. The missing map is `maps/connection_states`

### Describe how to test/QA your changes

- Start system-probe.
- For each map added, run the following:
```
sudo curl --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/ebpf_maps?maps=connection_states
```
Replace `<mapName>` by the name of the map you want to check. Beware that the newlines might be printed as `\n` in the output. You should see some output similar to this (here taking `connection_states` as an example):
```
Map: 'connection_states', key: '', value: '' \n(struct { Saddr_h uint64; Saddr_l uint64; Daddr_h uint64; Daddr_l uint64; Sport uint16; Dport uint16; Netns uint32; Pid uint32; Metadata uint32 }) {\n Saddr_h: (uint64) 0,\n Saddr_l: (uint64) 1077400330,\n Daddr_h: (uint64) 0,\n Daddr_l: (uint64) 37212938,\n Sport: (uint16) 22,\n Dport: (uint16) 53830,\n Netns: (uint32) 0,\n Pid: (uint32) 0,\n Metadata: (uint32) 1\n}\n(uint32) 2524529433\n(struct { Saddr_h uint64; Saddr_l uint64; Daddr_h uint64; Daddr_l uint64; Sport uint16; Dport uint16; Netns uint32; Pid uint32; Metadata uint32 }) {\n Saddr_h: (uint64) 0,\n Saddr_l: (uint64) 37212938,\n Daddr_h: (uint64) 0,\n Daddr_l: (uint64) 1077400330,\n Sport: (uint16) 53830,\n Dport: (uint16) 22,\n Netns: (uint32) 0,\n Pid: (uint32) 0,\n Metadata: (uint32) 1\n}\n(uint32) 2125272448\n(struct { Saddr_h uint64; Saddr_l uint64; Daddr_h uint64; Daddr_l uint64; Sport uint16; Dport uint16; Netns uint32; Pid uint32; Metadata uint32 }) {\n Saddr_h: (uint64) 0,\n Saddr_l: (uint64) 4272553641,\n Daddr_h: (uint64) 0,\n Daddr_l: (uint64) 1077400330,\n Sport: (uint16) 80,\n Dport: (uint16) 59148,\n Netns: (uint32) 0,\n Pid: (uint32) 0,\n Metadata: (uint32) 1\n}\n(uint32) 276044319\n"
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by guyarb** [comment]
regarding the "QA steps"
instead of 
`sudo curl --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/ebpf_maps|grep <mapName>`
you can use
`sudo curl --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/ebpf_maps?maps=<map name1>,<map name 2>,...`

File: pkg/network/usm/dump.go [hunk 1]
```diff
@@ -66,6 +76,25 @@ func (e *ebpfProgram) dumpMapsHandler(manager *manager.Manager, mapName string,
...
+		}
+
+	case protocolDispatcherClassificationPrograms: // maps/dispatcher_classification_progs (BPF_MAP_TYPE_PROG_ARRAY), key C.__u32, value C.__u32
+		output.WriteString("Map: '" + mapName + "', key: 'C.__u32', value: 'C.__u32'\n")
+		iter := currentMap.Iterate()
+		var key uint32
+		var value uint32
+		for iter.Next(unsafe.Pointer(&key), unsafe.Pointer(&value)) {
+			output.WriteString(spew.Sdump(key, value))
+		}
```
Comment (guyarb) [inline pkg/network/usm/dump.go:96]: That's a static map, a.k.a, it won't change during our execution of the program
I don't mind having it, but not sure if it has a real impact and value

File: pkg/network/usm/dump.go [hunk 2]
```diff
@@ -31,6 +32,15 @@ func (e *ebpfProgram) dumpMapsHandler(manager *manager.Manager, mapName string,
...
 		}
 
+	case protocols.ProtocolDispatcherProgramsMap: // maps/protocols_progs (BPF_MAP_TYPE_PROG_ARRAY), key C.__u32, value C.__u32
+		output.WriteString("Map: '" + mapName + "', key: 'C.__u32', value: 'C.__u32'\n")
+		iter := currentMap.Iterate()
+		var key uint32
+		var value uint32
+		for iter.Next(unsafe.Pointer(&key), unsafe.Pointer(&value)) {
+			output.WriteString(spew.Sdump(key, value))
+		}
```
Comment (guyarb) [inline pkg/network/usm/dump.go:42]: https://github.com/DataDog/datadog-agent/pull/17805/files#r1240668126

**Review by guyarb — APPROVED** [review]
Overall looks good.
Perhaps I should have been explicit in the jira ticket, but we have more maps to dump
like those exist in `pkg/network/ebpf/c/protocols/http/maps.h`

We can merge that PR, and prepare other PRs to dump the rest of the maps

File: pkg/network/usm/dump.go (hunk 1 above)
Comment (Yumasi) [reply pkg/network/usm/dump.go]: Removed it.

**Pull request merged**